operator/pkg/gateway-api/routechecks/gateway_checks.go

```diff
@@ -123,7 +121,14 @@ func CheckGatewayRouteKindAllowed(input Input, parentRef gatewayv1.ParentReferen
 				Message: routeGVK.Kind + " is not allowed to attach to this Gateway due to route kind restrictions",
 			})
 
-			return false, nil
+		} else {
+			input.SetParentCondition(parentRef, metav1.Condition{
+				Type:    string(gatewayv1.RouteConditionAccepted),
+				Status:  metav1.ConditionTrue,
+				Reason:  string(gatewayv1.RouteReasonAccepted),
+				Message: "Accepted " + routeGVK.Kind,
+			})
+			return true, nil
```

Comment:
In my understanding, if an HTTPRoute, without the `SectionName` field, is attached to the entire Gateway, then all listeners in that Gateway must allowed it for it to attach.

Reply:
Not quite. All Listeners that it _can_ attach to, it will. That means, that if the AllowedKinds do not match, or the hostname intersection doesn't work out, then the HTTPRoute will only attach to the valid Listeners.